src/cmd/promremotebench/scrape.go

```diff
@@ -99,18 +99,21 @@ func (g *gatherer) Gather() ([]*dto.MetricFamily, error) {
 				g.logger.Fatal("no metric family found for metric")
 			}
 
-			labels := make([]*dto.LabelPair, 0, len(series[i].Labels))
+			labelPairs := make([]*dto.LabelPair, 0, len(series[i].Labels))
 			for j := range series[i].Labels {
-				label := &dto.LabelPair{
-					Name:  &series[i].Labels[j].Name,
-					Value: &series[i].Labels[j].Value,
+				label := series[i].Labels[j]
```

Comment:
nit; Actually doesn't seem like you need to use `j` here, can probably just do `for _, label := range series[i].Labels{`  as well 👍

Reply:
I've actually tried that. The result was that the references to the changing `label` loop variable were stored in `LabelPair`, and all the `labelPairs` ended up with the same values after the loop ended.

Reply:
Ahhh I see because they're pointers right? I think you can probably get around it by explicitly capturing the variable with `label := label`, but that's not any nicer... would you mind adding a comment here about this just in case someone looks at this in future and tries to refactor it without realizing it would lead to issues? Alternatively if we can add a small case for this case

Reply:
@arnikola agreed. Added both a comment and a test. The test has actually revealed a similar issue in the loop below (fixed). Please have a look.